app/assets/javascripts/components/article_finder/article_finder.jsx

```diff
@@ -52,20 +55,38 @@ const ArticleFinder = createReactClass({
     }
   },
 
-  updateFields(key, value) {
-    return this.props.updateFields(key, value);
+  getQueryVariable() {
+    const query = window.location.search.substring(1);
+    const vars = query.split('&');
+    for (let i = 0; i < vars.length; i += 1) {
+        const pair = vars[i].split('=');
+        this.updateFields(pair[0], pair[1]);
+    }
+    },
+  async updateFields(key, value) {
+    await this.props.updateFields(key, value);
+    if (this.props.search_term.length !== 0) { this.buildURL(); }
   },
 
   toggleFilter() {
     return this.setState({
       showFilters: !this.state.showFilters
     });
   },
-
+  buildURL() {
+    let queryStringUrl = window.location.href.split('?')[0];
+    const search_term = encodeURIComponent(this.props.search_term);
+    const search_type = encodeURIComponent(this.props.search_type);
+    const article_quality = encodeURIComponent(this.props.article_quality);
+    const min_views = encodeURIComponent(this.props.min_views);
+    queryStringUrl += `?search_term=${search_term}&search_type=${search_type}&article_quality=${article_quality}&min_views=${min_views}`;
+    history.pushState(window.location.href, 'query_string', queryStringUrl);
```

Comment:
I think `pushState` is not what we want here; this fills up the browser history with every change. `replaceState` is probably right here.

Reply:
Got it ! Although had intentionally used .pushState() to save the searches to history if a user wants to retrieve old queries , but you are right , it's better not to clutter history with each and every change